src/lib/controller/store.test.ts

```diff
@@ -1,131 +1,73 @@
-  {
-    op: "add",
-    path: "add:/data/hello-pepr-a",
-    value: "secret",
-  },
-  {
-    op: "add",
-    path: "add:/data/hello-pepr-v2-b",
-    value: "secret",
-  },
-  {
-    op: "add",
-    path: "add:/data/hello-pepr-v2-c",
-    value: "secret",
-  },
-  {
-    op: "add",
-    path: "add:/data/hello-pepr-v2-d",
-    value: "secret",
-  },
-  {
-    op: "add",
-    path: "add:/data/hello-pepr-v2-e",
-    value: "secret",
-  },
-  {
-    op: "add",
-    path: "add:/data/hello-pepr-v2-f",
-    value: "secret",
-  },
-];
-
-describe("redactedPatch", () => {
   afterEach(() => {
-    delete process.env.PEPR_STORE_REDACT_VALUES;
+    jest.resetAllMocks();
+    jest.useRealTimers();
   });
 
-  test("should redact keys and values if PEPR_STORE_REDACT_VALUES is true", () => {
-    process.env.PEPR_STORE_REDACT_VALUES = "true";
-    addOperationKeys.forEach((key, i) => {
-      const redactedResult = redactedPatch({ [key]: addOperationValues[i] });
-      for (const [k, v] of Object.entries(redactedResult)) {
-        expect(k).toContain(`:${redactedValue}`);
-        expect(v).toEqual(expect.objectContaining({ value: redactedValue }));
-      }
+  describe("when initializing the store", () => {
+    beforeEach(() => {
+      testCapability = new Capability(capabilityConfig);
+      const mockPeprStore = new PeprStore();
+      const defaultMockImplementations = <T extends GenericClass, K extends KubernetesObject>() =>
+        ({
+          Patch: jest.fn().mockResolvedValueOnce(undefined as never),
+          InNamespace: jest.fn().mockReturnValueOnce({
+            Get: jest.fn().mockResolvedValueOnce(mockPeprStore as never),
+          }),
+          Watch: jest.fn().mockReturnValueOnce(undefined),
+          Apply: jest.fn().mockResolvedValueOnce(mockPeprStore as never),
+        }) as unknown as K8sInit<T, K>;
+
+      mockK8s.mockImplementationOnce(defaultMockImplementations);
+      jest.useFakeTimers();
     });
-  });
-  test("should not redact keys and values if PEPR_STORE_REDACT_VALUES is not true", () => {
-    addOperationKeys.forEach((key, i) => {
-      const redactedResult = redactedPatch({ [key]: addOperationValues[i] });
-      for (const [k, v] of Object.entries(redactedResult)) {
-        expect(k).not.toContain(`:${redactedValue}`);
-        expect(v).not.toEqual(expect.objectContaining({ value: redactedValue }));
-      }
+    it.skip("should migrate and setup the watch (with schedule)", async () => {
+      testCapability.OnSchedule(mockSchedule);
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const peprControllerStore = new PeprControllerStore([testCapability], `pepr-test-schedule`, () => {});
+      jest.advanceTimersToNextTimer();
+      await Promise.resolve();
+      expect(true).toBe(false); // store.ts only exposes a constructor, hard to test
+    });
+
+    it.skip("should create the store resource for a scheduled capability (without schedule)", async () => {
```

Comment:
```suggestion
    it("should create the store resource for a scheduled capability (without schedule)", async () => {
```

Reply:
This test doesn't do anything meaningful and is meant as a starting point for #1323.